docs/library/graphing/plotly.md

```diff
@@ -100,14 +44,82 @@ z_data = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/a
 fig = go.Figure(data=[go.Surface(z=z_data.values)])
 fig.update_traces(contours_z=dict(show=True, usecolormap=True,
                                   highlightcolor="limegreen", project_z=True))
-fig.update_layout(scene_camera_eye=dict(x=1.87, y=0.88, z=-0.64),
-                  width=500, height=500,
-                  margin=dict(l=65, r=50, b=65, t=90)
+fig.update_layout(
+    scene_camera_eye=dict(x=1.87, y=0.88, z=-0.64),
+    margin=dict(l=65, r=50, b=65, t=90)
 )
+
+def mountain_surface():
+    return rx.center(
+        rx.plotly(data=fig),
+    )
+```
+
+## Plot as State Var
+
+The figure need to be set as a state var if it needs to be updated during run time. 
+
+```python demo exec
+import plotly.express as px
+class PlotlyState(rx.State):
+    df = px.data.gapminder().query(f"country=='Canada'")
+    figure = px.line(
+        df,
+        x="year",
+        y="lifeExp",
+        title="Life expectancy in Canada",
+    )
+
+    def set_selected_country(self, country):
+        self.df = px.data.gapminder().query(f"country=='{country}'")
+        self.figure = px.line(
+            self.df,
+            x="year",
+            y="lifeExp",
+            title=f"Life expectancy in {country}",
+        )
+
+
+def line_chart_with_state():
+    return rx.vstack(
+        rx.select(
+            ['China', 'France', 'United Kingdom', 'United States', 'Canada'],
+            default_value="United States",
+            on_change=PlotlyState.set_selected_country,
+        ),
+        rx.plotly(data=PlotlyState.figure),
+    )
+```
+
+## Adding Styles and Layouts
+Use `update_layout()` method to update the layout of your chart. Checkout [Plotly Layouts](https://plotly.com/python/reference/layout/) for all layouts props. 
+
+```md alert info
+Note that the width and height props are not recommended to ensure the plot remains size responsive to its container. The size of plot will be determined by it's outer container. 
 ```
 
-Now pass the plotly figure again to the plotly component.
+```python demo exec
+df = px.data.gapminder().query("country=='Canada'")
+fig_1 = px.line(
+    df,
+    x="year",
+    y="lifeExp",
+    title="Life expectancy in Canada",
+)
+fig_1.update_layout(
+    title_x=0.5,
+    plot_bgcolor="#c3d7f7",
+    paper_bgcolor="rgba(128, 128, 128, 0.1)",
+    showlegend=True,
+    title_font_family="Open Sans",
```

Comment:
On my machine, the plot does not actually render with "Open Sans" title font. This looks like some default serif font.

![image](https://github.com/reflex-dev/reflex-web/assets/1524005/ab34e634-e666-471a-84db-c64dd6b6178e)

Reply:
Hmmm, does it have to do with need to have  Open Sans font installed on your system? If not it will render as default.